### ISSUE TITLE
Deserialization is inefficient

### ISSUE THREAD

**Issue by smashwilson** [atom/tree-view#1300]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Tree-view startup and deserialization takes a fair chunk of time. It's possible we're triggering a synchronous directory walk of the project structure.

### Steps to Reproduce

_N/A_

### Versions

_N/A_

### Additional Information

_N/A_

**Comment by 50Wliu** [comment]
Yeah, pretty much: https://github.com/atom/tree-view/blob/7a4b60cf765b2f1a86459571c7fb33fdeb7f3acb/lib/directory.js#L354 is called on every expanded directory, recursively.

**Comment by smashwilson** [comment]
😱
